api/consensus.go

```diff
@@ -25,25 +35,69 @@ func (srv *Server) consensusStatusHandler(w http.ResponseWriter, req *http.Reque
 		}
 	}
 
-	writeJSON(w, ConsensusInfo{
-		srv.blockchainHeight,
+	writeJSON(w, ConsensusSetStatus{
+		types.BlockHeight(srv.blockchainHeight),
 		srv.currentBlock.ID(),
 		currentTarget,
 	})
 }
 
-// consensusSynchronizeHandler handles the API call asking for the consensus to
-// synchronize with other peers.
-func (srv *Server) consensusSynchronizeHandler(w http.ResponseWriter, req *http.Request) {
+// consensusHandler handles the API calls to /consensus.
+func (srv *Server) consensusHandler(w http.ResponseWriter, req *http.Request) {
+	lockID := srv.mu.RLock()
+	defer srv.mu.RUnlock(lockID)
+
+	if req.Method == "" || req.Method == "GET" {
+		srv.consensusHandlerGET(w, req)
+		return
+	}
```

Comment:
ah, true. This is where using a real framework like [Martini](https://github.com/go-martini/martini) or [Gorilla](http://www.gorillatoolkit.org/pkg/mux) would be useful.